examples/systemcalls/systemcalls.c

```diff
@@ -93,6 +201,35 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
+    int status;
+    pid_t pid;
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if (fd < 0)
+    {
+    	syslog(LOG_ERR, "Error while opening file");
+    	return false;
+    }
+    pid=fork();
+    if(pid==-1)
+    	return false;
+    else if(pid == 0)
+    {
+    	if (dup2(fd, 1) < 0) 
+    	{
+    		syslog(LOG_ERR, "Error while executing dup2");
+    		return false;
+    	}
+    	close(fd);
+    	if((execv(full_path, command))!=-1)
+    		return true;
+    	else
+    		return false;
```

Comment:
@abaa3023 think about a way you can avoid duplicating logic here between the do_exec and do_exec_redirect cases.